tinytga/src/lib.rs

```diff
@@ -7,65 +7,97 @@
+//!
+//! The code uses the [`Tga`] struct and only works if the color format inside the TGA file is known
+//! at compile time. While this makes the code less flexible it offers the best performance by
+//! making sure that no unnecessary color conversions are used.
 //!
 //! ```rust
-//! use embedded_graphics::{prelude::*, pixelcolor::Rgb888};
-//! use tinytga::{Bpp, ImageOrigin, ImageType, RawPixel, Tga, TgaHeader};
+//! # fn main() -> Result<(), core::convert::Infallible> {
+//! # let mut display = embedded_graphics::mock_display::MockDisplay::default();
+//! use embedded_graphics::{image::Image, pixelcolor::Rgb888, prelude::*};
+//! use tinytga::Tga;
 //!
 //! // Include an image from a local path as bytes
 //! let data = include_bytes!("../tests/chessboard_4px_rle.tga");
 //!
-//! // Create a TGA instance from a byte slice.
-//! // The color type is set by defining the type of the `img` variable.
-//! let img: Tga<Rgb888> = Tga::from_slice(data).unwrap();
+//! let tga: Tga<Rgb888> = Tga::from_slice(data).unwrap();
 //!
-//! // Check the size of the image.
-//! assert_eq!(img.size(), Size::new(4, 4));
+//! let image = Image::new(&tga, Point::zero());
 //!
-//! // Collect pixels into a vector.
-//! let pixels: Vec<_> = img.pixels().collect();
+//! image.draw(&mut display)?;
+//! # Ok::<(), core::convert::Infallible>(()) }
 //! ```
 //!
-//! ## Drawing an image using `embedded-graphics`
+//! ## Using `DynamicTga` to draw an image
 //!
-//! This example demonstrates how a TGA image can be drawn to a [embedded-graphics] draw target.
+//! The previous example had the limitation that the color format needed to be known at compile
+//! time. In some use cases this can be a problem, for example if user supplied images should
+//! be displayed. To handle these cases [`DynamicTga`] can be used, which performs color conversion
+//! if necessary.
 //!
 //! ```rust
 //! # fn main() -> Result<(), core::convert::Infallible> {
-//! # let mut display = embedded_graphics::mock_display::MockDisplay::default();
+//! # let mut display = embedded_graphics::mock_display::MockDisplay::<Rgb888>::default();
 //! use embedded_graphics::{image::Image, pixelcolor::Rgb888, prelude::*};
-//! use tinytga::Tga;
+//! use tinytga::DynamicTga;
 //!
 //! // Include an image from a local path as bytes
 //! let data = include_bytes!("../tests/chessboard_4px_rle.tga");
-
-//! let tga: Tga<Rgb888> = Tga::from_slice(data).unwrap();
+//!
+//! let tga = DynamicTga::from_slice(data).unwrap();
 //!
 //! let image = Image::new(&tga, Point::zero());
 //!
 //! image.draw(&mut display)?;
 //! # Ok::<(), core::convert::Infallible>(()) }
 //! ```
+//! ## Accessing pixels using an embedded-graphics color type
+//!
+//! Even if tinytga is used without using [embedded-graphics] to draw the image the color types
+//! provided by [embedded-graphics] can still be used to access the pixel data using the
+//! [`pixels`](struct.Tga.html#method.pixels) method.
```

Comment:
This took a couple of readthroughs to understand the meaning. I've hopefully clarified it a bit:

```suggestion
//! If [embedded-graphics] is not used to draw the TGA image, the color types provided by 
//! [embedded-graphics] can still be used to access the pixel data using the 
//! [`pixels`](struct.Tga.html#method.pixels) method.
```